Packages/vcs/Lib/configurator.py

```diff
@@ -258,81 +199,124 @@ def detach(self):
             return
 
         if datetime.datetime.now() - self.clicking[1] < datetime.timedelta(0, .5):
-
             point = self.clicking[0]
             self.clicking = None
+
             if self.creating:
                 self.click_locations.append(point)
                 if len(self.click_locations) == CLICKS_TO_CREATE[self.creating]:
                     self.create()
                 return
 
-            if self.target and self.shift() == False and self.target.handle_click(point):
+            if self.target and self.shift() is False and self.target.handle_click(point):
                 return
 
             if self.shift() and type(self.target) != editors.group.GroupEditor:
                 self.target = editors.group.GroupEditor(self.interactor, (self.target,))
 
-            clicked = None
-            display_clicked = None
-            for display in self.displays:
-                obj = self.in_display_plot(point, display)
-                if obj is not None and (clicked is None or obj.priority >= clicked.priority):
-                    clicked = obj
-                    display_clicked = display
-            if clicked:
-                if self.target is None or self.target.is_object(clicked) == False:
-                    self.activate(clicked, display_clicked)
-                elif self.shift() and self.target.contains_object(clicked):
-                    self.target.remove_object(clicked)
-                    if len(self.target.targets) == 0:
-                        self.target.detach()
-                        self.target = None
-                        self.save()
-                        return
-            elif self.target is not None and self.shift() == False:
+            clicked_actor = self.actor_at_point(*point)
+
+            if clicked_actor is None and self.target:
                 self.deactivate(self.target)
                 return
 
+            if clicked_actor is None:
+                return
+
+            display, key = self.display_and_key_for_actor(clicked_actor)
+            if editable_type(display, key):
+                # Some methods (markers) have more than one actor per displayed item
+                if clicked_actor != display.backend[key] and clicked_actor not in display.backend[key]:
+                    for group in display.backend[key]:
+                        if clicked_actor in group:
+                            clicked_actor = group
+                            break
+
+                self.activate(display, clicked_actor, key)
+
         self.clicking = None
 
+    def display_and_key_for_actor(self, actor):
+        """
+        Iterates across the displays, and tries to find the actor in the backend.
+        """
+        for display in self.displays:
+            for key in display.backend:
+                try:
+                    if actor == display.backend[key] or actor in display.backend[key] or [True for group in display.backend[key] if actor in group]:
```

Comment:
I'd suggest manually wrapping lines longer than 80 characters. The author is better at breaking lines into easy-to-parse logical chunks than a text editor -- this make it easier to spot errors in logic, etc and quickly grok the conditional parameters while reading through the code.